### PR TITLE
feat: new rules related to ZScaler blog - OneNote: A Growing Threat for Malware Distribution

### PR DESCRIPTION
**Pull request by hieuttmmo**
### Summary of the Pull Request
Contribute new Sigma Rules to cover specific techniques mentioned by ZScaler in their recent blog post OneNote: A Growing Threat for Malware Distribution

### Detailed Description of the Pull Request / Additional Comments
Contribute 4 new rules
1. Suspicious Decoding Gzip Archive: Detects attempts of decoding encoded Gzip archive via PowerShell. 
2. PowerShell Web Download DLL File: Detects suspicious ways to download dll using PowerShell
3. Suspicious Rundll32 Execution With Image Extension: Detects execution of RUNDLL32.exe with DLL masquerading as image files
4. Qakbot Registry Key Activity: Detects registry key used by IceID in distributing malicious OneNote campaign
-->

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/sigmahq_conventions.md)

### CODE REVIEW

**Review by nasbench — APPROVED** [review]

File: rules/windows/registry/registry_event/registry_event_malware_qakbot_registry.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+date: 2023/03/13
+tags:
+    - attack.defense_evasion
+    - attack.t1112
+logsource:
+    category: registry_event
+    product: windows
+detection:
+    selection:
+        TargetObject|endswith: '\Software\firm\soft'
```
Comment (frack113) [inline rules/windows/registry/registry_event/registry_event_malware_qakbot_registry.yml:17]: I'm not a fan, as this path is very specific to this report, and is not monitored in sysmon configurations such as sysmon-modular,SwiftOnSecurity...

File: rules/windows/process_creation/proc_creation_win_powershell_decode_gzip.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+title: Gzip Archive Decode Via PowerShell
+id: 98767d61-b2e8-4d71-b661-e36783ee24c1
+status: experimental
+description: Detects attempts of decoding encoded Gzip archive via PowerShell.
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_powershell_decode_gzip.yml:4]:
```suggestion
description: Detects attempts of decoding encoded Gzip archives via PowerShell.
```

File: rules/windows/process_creation/proc_creation_win_powershell_decode_gzip.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    product: windows
+    category: process_creation
+detection:
+    selection:
+        CommandLine|contains|all:
+            - 'GZipStream'
+            - '::Decompress'
+    condition: selection
+falsepositives:
+    - Legitimate administrative scripts may use this functionality. Use "ParentImage" in combination with Script names and allowed users and applications to filter legitimate executions
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_powershell_decode_gzip.yml:19]:
```suggestion
    - Legitimate administrative scripts may use this functionality. Use "ParentImage" in combination with the script names and allowed users and applications to filter legitimate executions
```

File: rules/windows/process_creation/proc_creation_win_powershell_decode_gzip.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+level: medium
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_powershell_decode_gzip.yml:20]: Discussion: I am not sure about the medium level. This rule is more of a "feature used" tagging than actual indicator of maliciousness. Any thoughts?

File: rules/windows/process_creation/proc_creation_win_powershell_download_dll.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+title: Potential DLL File Download Via PowerShell Invoke-WebRequest
+id: 0f0450f3-8b47-441e-a31b-15a91dc243e2
+status: experimental
+description: Detects potential dll files download using PowerShell Invoke-WebRequest cmdlet
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_powershell_download_dll.yml:4]:
```suggestion
description: Detects potential DLL files being downloaded using the PowerShell Invoke-WebRequest cmdlet
```

File: rules/windows/process_creation/proc_creation_win_regsvr32_image.yml [hunk 6]
```diff
@@ -1,5 +1,8 @@
 title: Suspicious Regsvr32 Execution With Image Extension
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_regsvr32_image.yml:1]: please update the modified date

File: rules/windows/process_creation/proc_creation_win_regsvr32_image.yml [hunk 7]
```diff
@@ -21,21 +24,20 @@ detection:
         - OriginalFileName: '\REGSVR32.EXE'
     selection_cli:
         CommandLine|endswith:
-            - '.jpg'
-            # Add more image extension
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_regsvr32_image.yml:25]: please keep the comment

File: rules/windows/process_creation/proc_creation_win_regsvr32_image.yml [hunk 8]
```diff
@@ -1,5 +1,8 @@
 title: Suspicious Regsvr32 Execution With Image Extension
 id: 089fc3d2-71e8-4763-a8a5-c97fbb0a403e
+related:
+    - id: 4aa6040b-3f28-44e3-a769-9208e5feb5ec
+      type: similar
 status: experimental
 description: Detects execution of REGSVR32.exe with DLL masquerading as image files
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_regsvr32_image.yml:7]:
```suggestion
description: Detects the execution of REGSVR32.exe with DLL files masquerading as image files
```

File: rules/windows/process_creation/proc_creation_win_rundll32_susp_execution_with_image_extension.yml [hunk 9]
```diff
@@ -0,0 +1,41 @@
+title: Suspicious Rundll32 Execution With Image Extension
+id: 4aa6040b-3f28-44e3-a769-9208e5feb5ec
+related:
+    - id: 089fc3d2-71e8-4763-a8a5-c97fbb0a403e
+      type: similar
+status: experimental
+description: Detects execution of Rundll32.exe with DLL masquerading as image files
```
Comment (phantinuss) [inline rules/windows/process_creation/proc_creation_win_rundll32_susp_execution_with_image_extension.yml:7]:
```suggestion
description: Detects the execution of Rundll32.exe with DLL files masquerading as image files
```

File: rules/windows/registry/registry_event/registry_event_malware_qakbot_registry.yml [hunk 10]
```diff
@@ -0,0 +1,21 @@
+title: Potential Qakbot Registry Activity
+id: 1c8e96cd-2bed-487d-9de0-b46c90cade56
+status: experimental
+description: Detects registry key used by IceID in distributing malicious OneNote campaign
```
Comment (phantinuss) [inline rules/windows/registry/registry_event/registry_event_malware_qakbot_registry.yml:4]:
```suggestion
description: Detects a registry key used by IceID in a campaign that distributes malicious OneNote files
```

File: rules/windows/registry/registry_event/registry_event_malware_qakbot_registry.yml (hunk 1 above)
Comment (phantinuss) [reply rules/windows/registry/registry_event/registry_event_malware_qakbot_registry.yml:17]: I would even add it as `\Software\firm\soft\name` or at least as `\Software\firm\soft\` to be more specific. The issue with the sysmon-config is true, but shouldn't be a reason not to include it so Sigma. 

I would also increase the score to high or critical, if the registry path is specific enough.

File: rules/windows/process_creation/proc_creation_win_powershell_decode_gzip.yml (hunk 4 above)
Comment (phantinuss) [reply rules/windows/process_creation/proc_creation_win_powershell_decode_gzip.yml:20]: cc: @nasbench @frack113

File: rules/windows/process_creation/proc_creation_win_regsvr32_image.yml (hunk 6 above)
Comment (nasbench) [reply rules/windows/process_creation/proc_creation_win_regsvr32_image.yml:1]: I did that and no need it's re-ordering

File: rules/windows/registry/registry_event/registry_event_malware_qakbot_registry.yml (hunk 1 above)
Comment (nasbench) [reply rules/windows/registry/registry_event/registry_event_malware_qakbot_registry.yml:17]: - These are malware rules so it's okay for them to be specific. 
- These will be transferred to the DFIR folder in the future
- The key is specific as it's described in the report I will increase to high and see.

File: rules/windows/process_creation/proc_creation_win_powershell_decode_gzip.yml (hunk 4 above)
Comment (nasbench) [reply rules/windows/process_creation/proc_creation_win_powershell_decode_gzip.yml:20]: It's a "feature" that can be suspicious or benign hence it's medium. Low rules are more for enrichment and are very noisy (which doesn't apply necessarily here).

Once the hunting rule is public we will transfer these kinds of rules there.

**Review by phantinuss — APPROVED** [review]

File: rules/windows/registry/registry_event/registry_event_malware_qakbot_registry.yml [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+date: 2023/03/13
+tags:
+    - attack.defense_evasion
+    - attack.t1112
+logsource:
+    category: registry_event
+    product: windows
+detection:
+    selection:
+        TargetObject|endswith: '\Software\firm\soft'
```
Comment (phantinuss) [inline rules/windows/registry/registry_event/registry_event_malware_qakbot_registry.yml:17]:
```suggestion
        TargetObject|endswith: '\Software\firm\soft\'
```
at least add the `\`. If I didn't misread the report is has to be there as there is a subkey `\name`
image

**Review by phantinuss — APPROVED** [review]

File: rules/windows/registry/registry_event/registry_event_malware_qakbot_registry.yml (hunk 11 above)
Comment (nasbench) [reply rules/windows/registry/registry_event/registry_event_malware_qakbot_registry.yml:17]: image

**Pull request merged**